packages/app-desktop/gui/NoteEditor/NoteBody/CodeMirror/utils/useCursorUtils.test.ts

```diff
@@ -2,19 +2,64 @@ import { modifyListLines } from './useCursorUtils';
 
 describe('useCursorUtils', () => {
 
-	const listWithDashes = `- item1
-- item2
-- item3`;
+	const listWithDashes = [
+		'- item1',
+		'- item2',
+		'- item3',
+	];
 
-	const listNoDashes = `item1
-item2
-item3`;
+	const listWithNoPrefixes = [
+		'item1',
+		'item2',
+		'item3',
+	];
 
-	test('should remove "- " from beggining of each line of input string', () => {
-		expect(JSON.stringify(modifyListLines(listWithDashes.split('\n'), 0, '- '))).toBe(JSON.stringify(listNoDashes.split('\n')));
+	const listWithNumbers = [
+		'1. item1',
+		'2. item2',
+		'3. item3',
+	];
+
+	const listWithOnes = [
+		'1. item1',
+		'1. item2',
+		'1. item3',
+	];
+
+	const listWithSomeNumbers = [
+		'1. item1',
+		'item2',
+		'2. item3',
+	];
+
+	const listWithSomeNumbersAlternated = [
+		'item1',
+		'2. item2',
+		'item3',
+	];
+
+	test('should remove "- " from beginning of each line of input string', () => {
+		expect(modifyListLines([...listWithDashes], NaN, '- ')).toStrictEqual(listWithNoPrefixes);
 	});
 
-	test('should add "- " at the beggining of each line of the input string', () => {
-		expect(JSON.stringify(modifyListLines(listNoDashes.split('\n'), 0, '- '))).toBe(JSON.stringify(listWithDashes.split('\n')));
+	test('should add "- " at the beginning of each line of the input string', () => {
+		expect(modifyListLines([...listWithNoPrefixes], NaN, '- ')).toStrictEqual(listWithDashes);
 	});
+
+	test('should remove "n. " at the beginning of each line of the input string', () => {
+		expect(modifyListLines([...listWithNumbers], 4, '1. ')).toStrictEqual(listWithNoPrefixes);
+	});
+
+	test('should add "n. " at the beginning of each line of the input string', () => {
+		expect(modifyListLines([...listWithNoPrefixes], 1, '1. ')).toStrictEqual(listWithNumbers);
+	});
+
+	test('should remove "1. " at the beginning of each line of the input string', () => {
+		expect(modifyListLines([...listWithOnes], 2, '1. ')).toStrictEqual(listWithNoPrefixes);
+	});
+
+	test('should remove "n. " from each line that has it, and add it to the lines which do not', () => {
```

Comment:
I don't think this is the best behaviour because if the user selects a block, they want either all the numbers to be gone, or add numbers to each lines, but I don't think they want a mix of both. Perhaps you could check the first item - if it's a number, remove all numbers, and if there's no number, add a number to each line.

Reply:
Sure that makes sense. Should it be just the first line or any line?

Reply:
First line I guess?